### PR TITLE
port marker functions

### PR DESCRIPTION
**Pull request by brotzeit**

### CODE REVIEW

File: rust_src/src/marker.rs [hunk 1]
```diff
@@ -102,4 +107,63 @@ pub fn set_point_from_marker(marker: LispMarkerRef) {
...
+
+/// Set the insertion-type of MARKER to TYPE.
+/// If ITYPE is non-nil, it means the marker advances when you insert text at it.
+/// If ITYPE is nil, it means the marker stays behind when you insert text at it.
+#[lisp_fn]
+pub fn set_marker_insertion_type(mut marker: LispMarkerRef, itype: LispObject) -> LispObject {
+    unsafe { mset_insertion_type(marker.as_mut(), itype.is_not_nil() as BoolBF) };
+    itype
+}
+
```
Comment (shaleh) [inline rust_src/src/marker.rs]: `itype` is actually just a boolean then and this function returns a boolean.

File: rust_src/src/marker.rs [hunk 2]
```diff
@@ -102,4 +107,63 @@ pub fn set_point_from_marker(marker: LispMarkerRef) {
...
+/// The optional argument ITYPE specifies the insertion type of the new marker;
+/// see `marker-insertion-type'.
+#[lisp_fn(min = "0")]
+pub fn copy_marker(marker: LispObject, itype: LispObject) -> LispObject {
+    if marker.is_not_nil() {
+        marker.as_fixnum_coerce_marker_or_error();
+    }
+    let new = unsafe { LispObject::from_raw(Fmake_marker()) };
+    let buffer_or_nil = if marker.is_marker() {
+        marker_buffer(marker)
```
Comment (shaleh) [inline rust_src/src/marker.rs]: This no longer correct. `marker_buffer` now returns a `Option<Marker>`.

I think this should be:

buffer_or_nil = marker.as_marker().and_then(|m| m.buffer()).unwrap_or(LispObject::constant_nil(), |b| b.as_lisp_obj())

With proper formatting of course :-)

File: rust_src/src/marker.rs (hunk 1 above)
Comment (brotzeit) [reply rust_src/src/marker.rs]: No, it can be nil and non-nil. I also changed the docstring as it was wrong.

Comment (shaleh) [reply rust_src/src/marker.rs]: Looking at the lisp code, t and nil are the usual parameters.

Comment (shaleh) [reply rust_src/src/marker.rs]: Right. In Lisp NIL is false and anything else is true.

File: rust_src/src/marker.rs (hunk 2 above)
Comment (brotzeit) [reply rust_src/src/marker.rs]: Yeah, I have changed it. I should have made this one PR, sorry :)

Comment (shaleh) [reply rust_src/src/marker.rs]: No wories, it is easy to fix.

**Comment by shaleh** [comment]
Looks like transient network failure. Let's try again.

**Pull request merged**